### PR TITLE
fix: [CODE-431]: Update API endpoint for listing repos and move to getGeneralResponse

### PR DESCRIPTION
**Pull request by akhilpandey1**
## Describe your changes
- Fix API endpoint
- Introduce support for resource groups to be created with resource_identifier at project level
- Use getGeneralResponse instead of getResponse to bypass the harness response DTO requirement

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55120)

### CODE REVIEW

**Comment by akhilpandey1** [comment]
trigger messagecheck

**Review by abhinav-harness — DISMISSED** [review]

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55120)* status: 6 file(s) to review. 1 change requests. 

___
*[`platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/code/CodeRepositoryResourceImpl.java` line 94 at r2](https://reviewable.io/reviews/harness/harness-core/55120#-Niih0rX8ojru_nnCI6T:-Niih0rX8ojru_nnCI6U:b-3zfih9) ([raw file](https://github.com/harness/harness-core/blob/1830c45b5e1fd949b990f34ad99cd7e6f222885b/platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/code/CodeRepositoryResourceImpl.java#L94)):*
> ```Java
>     }
> 
>     List<CodeRepoResponseDTO> codeViewRepos = NGRestUtils.getGeneralResponse(codeViewRepoClient.getRepos(
> ```

Please elaborate on this change.

**Comment by akhilpandey1** [comment]
*[`platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/code/CodeRepositoryResourceImpl.java` line 94 at r2](https://reviewable.io/reviews/harness/harness-core/55120#-Niih0rX8ojru_nnCI6T:-NikkhQC3yog7U-cDnWY:b-299i6u) ([raw file](https://github.com/harness/harness-core/blob/1830c45b5e1fd949b990f34ad99cd7e6f222885b/platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/code/CodeRepositoryResourceImpl.java#L94)):*
<details><summary><i>Previously, Jimit09Gandhi (Jimit Gandhi) wrote…</i></summary><blockquote>

Please elaborate on this change.
</blockquote></details>

This is done because the usual NGRestUtils.getResponse() expects the data to be present inside the object under the `data` key, whereas harness-code doesn't follow this suite and sends the API data directly and it's not enclosed in an object as expected by previous call which was failing during validation.

**Review by Jimit09Gandhi — APPROVED** [review]
> Reviewed 3 of 6 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55120)* status: :shipit: complete! ✓

File: platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/ResourceTypeConstants.java [hunk 1]
```diff
@@ -103,4 +103,6 @@ public class ResourceTypeConstants {
   public static final String GITOPS_AGENT = "GITOPS_AGENT";
   public static final String GITOPS_PROJECT_MAPPING = "GITOPS_PROJECT_MAPPING";
   public static final String GITOPS_APPLICATION = "GITOPS_APPLICATION";
+
+  public static final String CODE_REPOSITORY = "CODE_REPOSITORY";
```
Comment (johannesHarness) [inline platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/ResourceTypeConstants.java:107]: this is exactly the same as we define in `CodeRepositoryResourceImpl` line 54. Do they have to match to work, and if so, should we have one source of truth. if not then ignore the comment

**Review by johannesHarness — DISMISSED** [review]

File: platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/ResourceTypeConstants.java (hunk 1 above)
Comment (akhilpandey1) [reply platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/ResourceTypeConstants.java:107]: Good catch, now we're referencing this in CodeRepositoryResourceImpl and not defining it again. Thanks

**Review by enver-bisevac — APPROVED** [review]
LGTM

**Review by Jimit09Gandhi — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55120)* status: 1. unresolved discussions.

**Pull request merged**